### PR TITLE
Propagate defStyleAttr to parent (and add 4th ctor)

### PR DESCRIPTION
**Pull request by ataulm**

### CODE REVIEW

File: app/src/main/java/com/example/defaultstyles/CustomCircleView.kt [hunk 1]
```diff
@@ -15,12 +15,18 @@ private const val DEFAULT_HAS_BORDER = false
 class CustomCircleView : View {
 
     constructor(context: Context) : this(context, null)
-    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, DEFAULT_STYLE_ATTR)
```
Comment (ataulm) [inline app/src/main/java/com/example/defaultstyles/CustomCircleView.kt:18]: If we don't pass this up to the `View` constructor, we don't give it the opportunity to apply any `View` properties.

Now, if we add a `View` (i.e. not custom) property to the default style, e.g. `android:padding` it'll be taken into account.

**Review by jshvarts — APPROVED** [review]
This makes sense! Thanks for the update

File: app/src/main/java/com/example/defaultstyles/CustomCircleView.kt [hunk 2]
```diff
@@ -15,12 +15,18 @@ private const val DEFAULT_HAS_BORDER = false
...
         attrs: AttributeSet?,
         defStyleAttr: Int
-    ) : super(context, attrs, defStyleAttr) {
+    ) : this(context, attrs, defStyleAttr, DEF_STYLE_RES)
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int,
+        defStyleRes: Int
+    ) : super(context, attrs, defStyleAttr, defStyleRes) {
```
Comment (ataulm) [inline app/src/main/java/com/example/defaultstyles/CustomCircleView.kt:29]: This one is only available from v21 I think, but same deal - without this, only the custom property (`CustomCircleView_hasBorder`) would be respected from the default style _res_ (the case where a default style attr is not specified).

File: app/src/main/java/com/example/defaultstyles/CustomCircleView.kt (hunk 1 above)
Comment (jshvarts) [reply app/src/main/java/com/example/defaultstyles/CustomCircleView.kt:18]: That's why I had trouble passing `android:color` :). Thanks!

**Pull request merged**